MapboxNavigation/Error.swift

```diff
@@ -70,28 +35,34 @@ public enum SpeechError: LocalizedError {
      The Speech API Did not successfully return a response.
      - parameter instruction: the instruction that failed.
      - parameter options: the SpeechOptions that were used to make the API request.
-     - parameter reason: a `SpeechRequestFailureReason` describing why the request failed.
      - parameter underlying: the underlying `Error` returned by the API.
      */
-    case apiRequestFailed(instruction: SpokenInstruction, options: SpeechOptions, reason: SpeechRequestFailureReason, underlying: Error?)
+    case apiError(instruction: SpokenInstruction, options: SpeechOptions, underlying: Error?)
+    
+    /**
+     The Speech API Did not return any data.
+     - parameter instruction: the instruction that failed.
+     - parameter options: the SpeechOptions that were used to make the API request.
+     */
+    case noData(instruction: SpokenInstruction, options: SpeechOptions)
     
     /**
      The speech engine was unable to perform an action on the system audio service.
      - parameter instruction: The instruction that failed.
      - parameter action: a `SpeechFailureAction` that describes the action attempted
-     - parameter engine: the `SpeechEngine` that tried to perform the action.
+     - parameter engine: the speech engine  that tried to perform the action.
      - parameter underlying: the `Error` that was optrionally returned by the audio service.
      */
-    case unableToControlAudio(instruction: SpokenInstruction?, action: SpeechFailureAction, engine: SpeechEngine, underlying: Error?)
+    case unableToControlAudio(instruction: SpokenInstruction?, action: SpeechFailureAction, engine: Any?, underlying: Error?)
     
     /**
      The speech engine was unable to initalize an audio player.
      - parameter playerType: the type of `AVAudioPlayer` that failed to initalize.
      - parameter instruction: The instruction that failed.
-     - parameter engine: The `SpeechEngine` that attempted the initalization.
+     - parameter engine: The speech engine that attempted the initalization.
      - parameter underlying: the `Error` that was returned by the system audio service.
      */
-    case unableToInitalizePlayer(playerType: AVAudioPlayer.Type, instruction: SpokenInstruction, engine: SpeechEngine, underlying: Error)
+    case unableToInitializePlayer(playerType: AVAudioPlayer.Type, instruction: SpokenInstruction, engine: Any?, underlying: Error)
```

Comment:
“Engine” is fairly odd to see by itself like this. I get that it comes from “text-to-speech engine”, but the iOS term for a TTS engine is “speech synthesizer”, so “synthesizer” would be a better label to use here.